cmd/kubeadm/app/discovery/file/file.go

```diff
@@ -131,11 +135,18 @@ func ValidateConfigInfo(config *clientcmdapi.Config, clustername string, discove
 		return config, nil
 	}
 
-	refreshedCluster := kubeconfigutil.GetClusterFromKubeConfig(refreshedBaseKubeConfig)
-	currentCluster.Server = refreshedCluster.Server
-	currentCluster.CertificateAuthorityData = refreshedCluster.CertificateAuthorityData
+	_, refreshedCluster := kubeconfigutil.GetClusterFromKubeConfig(refreshedBaseKubeConfig)
+	if currentCluster.Server != refreshedCluster.Server {
+		klog.Warningf("[discovery] the API Server endpoint %q in use is different from the endpoint %q which defined in the %s ConfigMap", currentCluster.Server, refreshedCluster.Server, bootstrapapi.ConfigMapClusterInfo)
+	}
+
+	if len(currentCluster.CertificateAuthorityData) == 0 && len(refreshedCluster.CertificateAuthorityData) > 0 {
+		// Now that we know the cluster CA, and the InsecureSkipTLSVerify can be disabled
+		config.Clusters[currentClusterName].CertificateAuthorityData = refreshedCluster.CertificateAuthorityData
+		config.Clusters[currentClusterName].InsecureSkipTLSVerify = false
```

Comment:
might be a good idea to just not update the InsecureSkipTLSVerify based on ca contents